requirements_doc.txt

```diff
@@ -1,6 +1,6 @@
 breathe==4.36.0
 docutils==0.21.2
 recommonmark==0.7.1
-sphinx>7.0.0,=<8.2.3
```

Comment:
Please break up this commit into three.  One should have this requirements fix.  The 2nd should have the *.c file changes.  The final one should have the updated test file.

Also, your commit message has a first line that is too long.  Note on GitHub how the first line is shown as a title for this commit.  With a long first line the bold title is truncated oddly.  Just have short first lines, and add additional info in the 2nd line in the commit.